Bad typing preventing typescript compilation without skipLibCheck

Bringing in plop as a dependency to a typescript project that currently does not use "skipLibCheck" fails compilation with the following error:

../../node_modules/node-plop/types/index.d.ts(142,3): error TS2502: 'inquirer' is referenced directly or indirectly in its own type annotation.
../../node_modules/node-plop/types/index.d.ts(142,22): error TS2694: Namespace '"../node_modules/@types/inquirer/index"' has no exported member 'Inquirer'.

This forces users to use "skipLibCheck" or not be able to use this library.

Since upgrading from typescript 4.7.4 to typescript 4.8.4, have a different issue with similar root cause :

node_modules/@types/inquirer/index.d.ts(80,76): error TS2344: Type 'T' does not satisfy the constraint 'Answers'.
node_modules/@types/inquirer/index.d.ts(84,91): error TS2344: Type 'TChoiceMap' does not satisfy the constraint 'Answers'.
node_modules/@types/inquirer/index.d.ts(84,105): error TS2344: Type 'T' does not satisfy the constraint 'Answers'.
node_modules/@types/inquirer/index.d.ts(139,43): error TS2344: Type 'T' does not satisfy the constraint 'Answers'.
node_modules/@types/inquirer/lib/objects/choices.d.ts(11,39): error TS2344: Type 'T' does not satisfy the constraint 'Answers'.
node_modules/@types/inquirer/lib/objects/choices.d.ts(11,61): error TS2344: Type 'T' does not satisfy the constraint 'Answers'.

Even with skipLibCheck build fail because of this.

--- a/packages/node-plop/types/index.d.ts
+++ b/packages/node-plop/types/index.d.ts
@@ -1,8 +1,10 @@
 import inquirer from "inquirer";
+
+type Inquirer = typeof inquirer;
+
 // @types/globby doesn't export types for GlobOptions, so we have to work a little bit to extract them:
 // GlobOptions is the second parameter of the sync function, which can be extracted with the Parameters<T> type
 import { globbySync } from "globby";
-
 type GlobOptions = Parameters<typeof globbySync>[1];
 import { HelperDelegate as HelperFunction } from "handlebars";
 
@@ -139,7 +141,7 @@ export type PromptQuestion =
   | inquirer.InputQuestion;
 
 export type DynamicPromptsFunction = (
-  inquirer: inquirer.Inquirer
+  inquirer: Inquirer
 ) => Promise<inquirer.Answers>;
 export type DynamicActionsFunction = (data?: inquirer.Answers) => ActionType[];
 

